src/Covenant/Internal/Unification.hs

```diff
@@ -40,7 +40,7 @@ data TypeAppError
   | -- | A wildcard (thus, a skolem) escaped its scope.
     LeakingWildcard Word64 Int (Index "tyvar")
   | -- | We were given too many arguments.
-    ExcessArgs (Vector (ValT Renamed))
+    ExcessArgs (Vector (Maybe (ValT Renamed)))
   | -- | We weren't given enough arguments.
     InsufficientArgs
```

Comment:
Minor point: Can we return the _function_ type for the thing that has an insufficient # arguments? I noticed while running into issues debugging the renamer that error messages without any kind of indication of the problem make debugging a lot harder. Maybe there's something else we could note in the error? I dunno, I just feel like I'll run into this and _any_ kind of information to point me at the cause will help a lot.

Reply:
Yeah, that's an easy enough fix I think.